### PR TITLE
Extend CurrentWorkflowConditionFailedError to include more information about current workflow

### PR DESCRIPTION
**Pull request by wxing1292**
**What changed?**
* Extend CurrentWorkflowConditionFailedError to include
  * RequestID
  * RunID
  * State
  * LastWriteVersion

**Why?**
Provide more info for the error, next PR will deprecate WorkflowExecutionAlreadyStartedError in favor of CurrentWorkflowConditionFailedError

**How did you test it?**
Existing tests

**Potential risks**
N/A

**Is hotfix candidate?**
No

### CODE REVIEW

File: common/persistence/cassandra/errors.go [hunk 1]
```diff
@@ -199,11 +201,27 @@ func extractCurrentWorkflowConflictError(
 
 	actualCurrentRunID := gocql.UUIDToString(record["current_run_id"])
 	if actualCurrentRunID != requestCurrentRunID {
+		binary, _ := record["execution_state"].([]byte)
+		encoding, _ := record["execution_state_encoding"].(string)
+		executionState := &persistencespb.WorkflowExecutionState{}
+		if state, err := serialization.WorkflowExecutionStateFromBlob(binary, encoding); err != nil {
```
Comment (yiminc) [inline common/persistence/cassandra/errors.go:207]: understand that we need this call here. but we were hoping that we could eliminate encoding/decoding here at this layer? is it reasonable to bubble up the blob from this layer and let upper layer do decoding?

File: common/persistence/cassandra/errors.go [hunk 2]
```diff
@@ -199,11 +201,27 @@ func extractCurrentWorkflowConflictError(
...
+			// noop, this means execution state cannot be parsed
+		} else {
+			executionState = state
+		}
```
Comment (yiminc) [inline common/persistence/cassandra/errors.go:211]: isn't it more clear to do just:
if ...; err == nil {
  executionState = state
}

File: common/persistence/sql/execution.go [hunk 3]
```diff
@@ -145,13 +146,21 @@ func (m *sqlExecutionStore) createWorkflowExecutionTx(
 					Msg: fmt.Sprintf("Workflow execution creation condition failed. WorkflowId: %v, "+
 						"LastWriteVersion: %v, PreviousLastWriteVersion: %v",
 						workflowID, row.LastWriteVersion, request.PreviousLastWriteVersion),
+					RequestID:        row.CreateRequestID,
+					RunID:            row.RunID.String(),
+					State:            row.State,
```
Comment (yiminc) [inline common/persistence/sql/execution.go:151]: oh, did we store decoded version of state for SQL?

File: common/persistence/cassandra/errors.go (hunk 1 above)
Comment (wxing1292) [reply common/persistence/cassandra/errors.go:207]: >is it reasonable to bubble up the blob from this layer and let upper layer do decoding?

definitely, the serialization / deserialization should not be done at this layer, but for the sake of this internal error migration / deprecation, this need to happen

this PR aims to get rid of this
https://github.com/temporalio/temporal/blob/21aa499c2e9ab0689572a425a52e977b0f0cc2a6/common/persistence/cassandra/cassandraPersistence.go#L937-L1056

File: common/persistence/cassandra/errors.go (hunk 2 above)
Comment (wxing1292) [reply common/persistence/cassandra/errors.go:211]: sure i can to the above

File: common/persistence/sql/execution.go (hunk 3 above)
Comment (wxing1292) [reply common/persistence/sql/execution.go:151]: yep

**Review by yiminc — APPROVED** [review]

**Pull request merged**